Adds `app open` command - Closes #2940

martinlingstuyl opened the pull request:
Closes #2940

Returns deep link to open Azure portal on the page to manage the current AAD app registration. Allows for triggering the preview portal and allows for automatically opening the browser.

waldekmastykarz commented on src/m365/aad/commands/app/app-open.ts:
```diff
@@ -0,0 +1,123 @@
+  public options(): CommandOption[] {
+    const options: CommandOption[] = [
+      { option: '--appId <appId>' },
+      { option: '--preview [preview]' },
```
`--preview` is meant to be used as a flag, so we shouldn't include `[preview]`

waldekmastykarz commented on src/m365/aad/commands/app/app-open.ts:
```diff
@@ -0,0 +1,123 @@
+    const options: CommandOption[] = [
+      { option: '--appId <appId>' },
+      { option: '--preview [preview]' },
+      { option: '--autoOpenBrowser [autoOpenBrowser]' }
```
`--autoOpenBrowser` is meant to be used as a flag so we shouldn't include `[autoOpenBrowser]`

waldekmastykarz commented on docs/docs/cmd/aad/app/app-open.md:
```diff
@@ -0,0 +1,42 @@
+## Options
+
+`--appId <appId>`
+: Application (client) ID of the Azure AD application registration to get.
```
`...to open`, as we're not retrieving the information about the AAD app using this command

waldekmastykarz commented on docs/mkdocs.yml:
```diff
@@ -31,6 +31,7 @@ nav:
         - app delete: 'cmd/aad/app/app-delete.md'
         - app get: 'cmd/aad/app/app-get.md'
         - app set: 'cmd/aad/app/app-set.md'
+        - app open: 'cmd/aad/app/app-open.md'
```
Let's sort the TOC alphabetically to make it easier for our users to find commands

waldekmastykarz commented on src/m365/aad/commands.ts:
```diff
@@ -5,6 +5,7 @@ export default {
   APP_DELETE: `${prefix} app delete`,
   APP_GET: `${prefix} app get`,
   APP_SET: `${prefix} app set`,
+  APP_OPEN: `${prefix} app open`,
```
Let's sort commands alphabetically for easier lookup

waldekmastykarz reviewed (changes requested):
Looks good! Let's do a few small adjustments before we proceed with a review

waldekmastykarz commented on docs/docs/cmd/aad/app/app-open.md:
```diff
@@ -0,0 +1,42 @@
+`--preview`
+: Use to open the url of the Azure AD preview portal.
+
+`--autoOpenBrowser`
```
Based on what we discussed in https://github.com/pnp/cli-microsoft365/issues/3118, shall we remove this option in favor of the global config setting?

martinlingstuyl replied:
Jep, I think that would be best. I will update the code and push it for review again.

waldekmastykarz replied:
Awesome! Thank you 👍

martinlingstuyl replied:
Oh wait, the setting is still called `autoOpenBrowserOnLogin`. I think I should wait until it is renamed. Is that already in the pipeline?

waldekmastykarz replied:
I think that task is still open 😬

martinlingstuyl replied:
In that case let's just keep this pull request open until it is. I'm in no hurry 😊

martinlingstuyl replied:
...or let me fix it myself 😂

waldekmastykarz replied:
> Based on what we discussed in #3118, shall we remove this option in favor of the global config setting?

Haven't we agreed to remove this option?

martinlingstuyl commented:
Hi guys, I refactored this one to use the new global setting.

waldekmastykarz commented:
Awesome! We'll review it shortly 👏

waldekmastykarz reviewed (changes requested):
I think we might be looking at the wrong issue. #2940 is for `app open` rather than `aad app open` and is meant to pick up the app ID from the .m365rc.json file rather than expect user to specify it.

martinlingstuyl commented:
> I think we might be looking at the wrong issue. #2940 is for `app open` rather than `aad app open` and is meant to pick up the app ID from the .m365rc.json file rather than expect user to specify it.

Ah, I see. 😁
I created the file in the wrong place, and missed the integration with `.m365rc.json`. 

I will be moving the fuctionality and fixing the integration with `.m365rc.json`.  
Can you elaborate on how you meant the  `appId` parameter to be used?

waldekmastykarz commented:
> Can you elaborate on how you meant the appId parameter to be used?

In the .m365rc.json file you can store information about multiple AAD apps. If you have multiple apps stored, and haven't specified which app you want to open, CLI will prompt you to choose one app. You can suppress this prompt by specifying the ID of the app which you'd like to view using the `appId` property.

martinlingstuyl commented:
Ok, great, I'll implement it like that

martinlingstuyl commented:
Ok, I was behind on this one as well. I moved it to the correct place and updated the functionality to actually do what it should have done in the first place. 🚀🚀

Hope it's okay like this.

plamber commented:
Manually merged. Thank you
